### PR TITLE
Add sysroot shortcut to rust-project.json


### PR DESCRIPTION
**Pull request by matklad**
bors r+
🤖

### CODE REVIEW

**Comment by matklad** [comment]
https://github.com/rust-analyzer/rust-project.json-example is updated to take advantage of this.

File: docs/user/manual.adoc [hunk 1]
```diff
@@ -273,9 +273,18 @@ However, if you use some other build system, you'll have to describe the structu
 [source,TypeScript]
 ----
 interface JsonProject {
-   /// The set of crates comprising the current project.
-   /// Must include all transitive dependencies as well as sysroot crate (libstd, libcore and such).
-   crates: Crate[];
+    /// Path to the directory with *source code* of sysroot crates.
+    ///
+    /// It should point to dircotry where std, core, and friends can be found:
```
Comment (jamesmunns) [inline docs/user/manual.adoc:278]:
```suggestion
    /// It should point to the directory where std, core, and friends can be found:
```

File: docs/user/manual.adoc [hunk 2]
```diff
@@ -273,9 +273,18 @@ However, if you use some other build system, you'll have to describe the structu
...
+    /// https://github.com/rust-lang/rust/tree/master/library.
+    ///
+    /// If provided, rust-analyzer automatically adds dependencies on sysroot crates.
+    /// Conversly, if you omit this path, you can specify sysroot dependencies yourself and,
```
Comment (jamesmunns) [inline docs/user/manual.adoc:282]:
```suggestion
    /// Conversely, if you omit this path, you can specify sysroot dependencies yourself and,
```

**Pull request merged**